File: Tests/OrbeEditorCoreTests/Fixtures/samples/sample.py
"""Line index over UTF-16 offsets."""

class LineIndex:
    def __init__(self, text: str) -> None:
        self.starts = [0]
        for i, ch in enumerate(text):
            if ch == "\n":
                self.starts.append(i + 1)

    def point(self, offset: int) -> tuple[int, int]:
        row = max(i for i, s in enumerate(self.starts) if s <= offset)
        return row, offset - self.starts[row]

File: Tests/OrbeEditorCoreTests/Fixtures/samples/sample.rs
//! Line index over byte offsets.

pub struct LineIndex {
    starts: Vec<usize>,
}

impl LineIndex {
    pub fn new(text: &str) -> Self {
        let mut starts = vec![0];
        for (i, b) in text.bytes().enumerate() {
            if b == b'\n' {
                starts.push(i + 1);
            }
        }
        Self { starts }
    }

    pub fn line_count(&self) -> usize {
        self.starts.len()
    }
}

fn main() {
    println!("{}", LineIndex::new("a\nb").line_count());
}
